prog/vm/github_runner.rb

```diff
@@ -287,22 +287,31 @@ def setup_info
 
     hop_wait
   rescue Octokit::Conflict => e
-    unless e.message.include?("Already exists")
-      raise e
-    end
+    raise e unless e.message.include?("Already exists")
+
     # If the runner already exists at GitHub side, this suggests that the
-    # process terminated prematurely before start the runner script and hop wait.
-    # We need to locate the 'runner_id' using the name and delete it.
-    # After this, we can register the runner again.
+    # process terminated prematurely before hop wait. We can't be sure if the
+    # script was started or not without checking the runner status. We need to
+    # locate the runner using the name and decide delete or continue to wait.
     runners = github_client.paginate("/repos/#{github_runner.repository_name}/actions/runners") do |data, last_response|
       data[:runners].concat last_response.data[:runners]
     end
     unless (runner = runners[:runners].find { _1[:name] == github_runner.ubid.to_s })
       fail "BUG: Failed with runner already exists error but couldn't find it"
     end
-    Clog.emit("Deleting GithubRunner because it already exists") { {github_runner: github_runner.values.merge({runner_id: runner[:id]})} }
-    github_client.delete("/repos/#{github_runner.repository_name}/actions/runners/#{runner[:id]}")
-    nap 5
+
+    # If the runner script is not started yet, we can delete the runner and
+    # register it again.
+    if vm.sshable.cmd("systemctl show -p SubState --value #{SERVICE_NAME}").chomp == "dead"
+      Clog.emit("Deleting GithubRunner because it already exists") { {github_runner: github_runner.values.merge({runner_id: runner[:id]})} }
+      github_client.delete("/repos/#{github_runner.repository_name}/actions/runners/#{runner[:id]}")
+      nap 5
+    end
+
+    # The runner script is already started. We persist the runner_id and allow
+    # wait label to decide the next step.
+    github_runner.update(runner_id: runner[:id], ready_at: Time.now)
```

Comment:
```suggestion
    github_runner.update(runner_id: runner.fetch(:id), ready_at: Time.now)
```

presuming there is no useful application of a "nil" here

Reply:
`runner[:id]` was called three times. I replaced it with `runner_id = runner.fetch(:id)`.